recipes/orc/all/conanfile.py

```diff
@@ -0,0 +1,157 @@
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        VirtualBuildEnv(self).generate()
+        VirtualRunEnv(self).generate(scope="build")
+
+        tc = CMakeToolchain(self)
+        tc.variables["ORC_PACKAGE_KIND"] = "conan"
+        tc.variables["BUILD_JAVA"] = False
+        tc.variables["BUILD_CPP_TESTS"] = False
+        tc.variables["BUILD_TOOLS"] = self.options.build_tools
+        tc.variables["BUILD_LIBHDFSPP"] = False
+        tc.variables["BUILD_POSITION_INDEPENDENT_LIB"] = bool(self.options.get_safe("fPIC", True))
+        tc.variables["INSTALL_VENDORED_LIBS"] = False
+        # AVX512 support is determined by ORC_USER_SIMD_LEVEL env var at runtime, defaults to off
+        tc.variables["BUILD_ENABLE_AVX512"] = self.options.get_safe("build_avx512", False)
+        tc.variables["STOP_BUILD_ON_WARNING"] = False
+        tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
+
+        # CMake versions less than 3.12 are supported by ORC pre-1.9.0 versions.
+        # Setting policy CMP0077 to NEW to remove unnecessary cache_variables settings.
+        if self.version < "1.9.0":
+            tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0077"] = "NEW"
+
+        protoc_path = os.path.join(self.dependencies["protobuf"].cpp_info.bindir, "protoc")
+        tc.variables["PROTOBUF_EXECUTABLE"] = protoc_path.replace("\\", "/")
+        tc.variables["HAS_POST_2038"] = self.settings.os != "Windows"
+        tc.variables["HAS_PRE_1970"] = self.settings.os != "Windows"
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def _patch_sources(self):
+        if self._should_patch_thirdparty_toolchain:
+            replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                            "ThirdpartyToolchain", "ConanThirdpartyToolchain")
+        # Allow shared builds
+        replace_in_file(self, os.path.join(self.source_folder, "c++", "src", "CMakeLists.txt"),
+                        "add_library (orc STATIC ${SOURCE_FILES})", "add_library (orc ${SOURCE_FILES})")
+
+    def build(self):
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        copy(self, "NOTICE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        if self.settings.os == "Windows" and self.options.shared:
+            mkdir(self, os.path.join(self.package_folder, "bin"))
+            os.rename(os.path.join(self.package_folder, "lib", "orc.dll"),
+                      os.path.join(self.package_folder, "bin", "orc.dll"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["orc"]
```

Comment:
```suggestion
        self.cpp_info.libs = ["orc"]
        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.system_libs = ["pthread", "m"]
```

This project uses pthread and libm. It's working in test package because is getting a ride from dependencies, but each recipe should be explicit about what is needed.

- https://github.com/apache/orc/blob/main/c%2B%2B/src/CpuInfoUtil.cc#L47
- https://github.com/apache/orc/blob/main/c%2B%2B/src/BloomFilter.hh#L25